core/src/banking_stage/scheduler_messages.rs

```diff
@@ -0,0 +1,76 @@
+use {
+    crate::immutable_deserialized_packet::ImmutableDeserializedPacket,
+    solana_sdk::transaction::SanitizedTransaction, std::sync::Arc,
+};
+
+/// A unique identifier for a transaction batch.
+/// The lower 64 bits are based on the order the transaction batch was scheduled.
+/// The upper 64 bits are an optional priority field - currently unused.
+#[derive(Clone, Copy, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TransactionBatchId(u128);
+
+impl TransactionBatchId {
+    pub fn new(index: u64) -> Self {
+        Self(index as u128)
+    }
+}
+
+/// A unique identifier for a transaction.
+/// The lower 64 bits are based on the order the transaction entered banking stage.
+/// The upper 64 bits are the priority of the transaction.
+#[derive(Clone, Copy, Debug, Default, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TransactionId(u128);
+
+impl TransactionId {
+    pub fn new(priority: u64, index: u64) -> Self {
+        Self(((priority as u128) << 64) | (index as u128))
+    }
+}
+
+/// Message: [Scheduler -> Worker]
+/// Transactions to be consumed (executed, recorded, committed)
+pub struct ConsumeWork {
+    pub batch_id: TransactionBatchId,
+    pub transaction_ids: Vec<TransactionId>,
+    pub transactions: Vec<SanitizedTransaction>,
+}
+
+/// Message: [Worker -> Scheduler]
+/// Transactions to be forwarded to the next leader(s)
+pub struct ForwardWork {
+    pub ids: Vec<TransactionId>,
+    pub packets: Vec<Arc<ImmutableDeserializedPacket>>,
```

Comment:
Mmm, not too sure about this asymmetry: Scheduler pushes down Vec<SanitizedTransaction>, but have `packets` on reversed direction.

Reply:
Could you elaborate what you mean on the assymetry?

I've added a diagram in the PR description with intended message flow, but it's summarized as:
* Every `ConsumeWork` sent from Scheduler to Worker is matched with a `FinishedConsumeWork` send from Worker to Scheduler
* Every `ForwardWork` sent from Scheduler to Worker is matched with a `FinishedForwardWork` send from Worker to Scheduler

Reply:
Oops, sorry for the incomplete sentence. By asymmetry I mean `vec<SanitizedTransactions>` go from Scheduler to Worker, but `Vec<Arc<ImmutableDeserializedPacket>>` back from Worker to Scheduler. Would be possible to pass either Transactions or Packets on both direction? In another words, transaction<->Packet conversion to be hold at one party.

Reply:
Maybe I'm still misunderstanding

The scheduler is the party responsible for transaction <-> packet conversions.
If the scheduler sends `Vec<SanitizedTransaction>` in a `ConsumeWork`, it will receive back `Vec<SanitizedTransaction>` in a `FinishedConsumeWork`.
If the scheduler sends `Vec<Arc<ImmutableDeserializedPacket>>` in a `ForwardWork`, it will receive back a `Vec<Arc<ImmutableDeserializedPacket>>` in a `FinishedForwardWork`.

The way I approached it, is the `Worker` is just given work to do. For consuming, we need `SanitizedTransaction` so that's what is passed in the messages. For forwarding, we need packets, so I pass the `Arc<ImmutableDeserializedPacket>` (serialization to actual packets still happens at the `Worker`).

Reply:
Was thinking maybe both messages can use same type, either Transaction or Packet, so all Workers will take Transaction, make interface more generic. But I can see you reasoning here.